week03/inho/java-christmas-6/src/main/java/christmas/validator/MenuValidator.java

```diff
@@ -0,0 +1,120 @@
+        orders = new LinkedHashMap<>();
+        menus = new ArrayList<>();
+
+        otherString = otherString.replaceAll(" ", "");
+
+        return validOtherString(otherString) && validContainUsingMenu(orders) &&
+                validLessMinOverMax(menus) && validOrderOnlyDrink(menus);
+    }
+
+    private boolean validOtherString(String otherString) {
+        try {
+            parsingAndSettingOrderString(otherString);
+        } catch (IllegalArgumentException e) {
+            errorMessage = ErrorMessage.ERROR_ORDER;
+
+            return false;
+        }
+        return true;
+    }
+
+    private void parsingAndSettingOrderString(String otherString) {
+        List<String> orderStrings = Arrays.stream(otherString.split(",", SPLIT_LIMIT)).toList();
+        orderStrings.forEach(this::settingOrderMap);
+    }
+
+    private void settingOrderMap(String orderString) {
+        String[] order = orderString.split("-", SPLIT_LIMIT);
+        int size = Integer.parseInt(order[1]);
+
+        if (size <= 0)
+            throw new IllegalArgumentException();
+
+        if (orders.containsKey(order[0]))
+            throw new IllegalArgumentException();
+
+        orders.put(order[0], size);
+    }
+
+    private boolean validContainUsingMenu(HashMap<String, Integer> orders) {
+        try {
+            orders.forEach((name, size) -> menus.add(new Menu(name, size)));
+        } catch (IllegalArgumentException e) {
+            errorMessage = ErrorMessage.ERROR_ORDER;
+
+            return false;
+        }
+        return true;
+    }
+
+    private boolean validLessMinOverMax(List<Menu> menus) {
+        int size = calculateTotalOrderSize(menus);
+
+        try {
+            if (size < MIN_ORDER || size > MAX_ORDER) {
+                throw new IllegalArgumentException();
+            }
+        } catch (IllegalArgumentException e) {
+            errorMessage = ErrorMessage.ERROR_OVER_MAX_SIZE;
+
+            return false;
+        }
+        return true;
+    }
+
+    private int calculateTotalOrderSize(List<Menu> menus) {
+        return menus.stream()
+                .mapToInt(Menu::getSize)
+                .sum();
+    }
+
+    private boolean validOrderOnlyDrink(List<Menu> menus) {
+        try {
+            if (isAllDrinkMenu(menus)) {
+                throw new IllegalArgumentException();
+            }
+        } catch (IllegalArgumentException e) {
+            errorMessage = ErrorMessage.ERROR_OVER_ONLY_DRINK;
+
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isAllDrinkMenu(List<Menu> menus) {
+        return menus.stream()
+                .allMatch(menu -> menu.checkManuKind(MenuKind.DRINK));
+    }
+
+
+    public List<Menu> getMenus() {
+        return Collections.unmodifiableList(menus);
+    }
+
+    public String getErrorMessage() {
+        if (errorMessage == null){
+            return "";
+        }
+
+        return errorMessage.toString();
+    }
```

Comment:
```suggestion
    private static final String EMPTY_MESSAGE = "";

    public String getErrorMessage() {
        if (errorMessage == null){
            return EMPTY_MESSAGE;
        }

        return errorMessage.toString();
    }
```

- "" (빈 문자열)을 반환하는 것보다는 변경 용이성, 명확한 의미 전달을 위해 상수 문자열로 만들어 사용하는 것도 좋을 것 같아요!

Reply:
확실히 지금은 숫자만 의미를 부여한다고 생각했습니다. 빈 문자열에도 의미를 부여해 보도록 하겠습니당